tests/core/filtering/test_basic_filter_tests.py

```diff
@@ -51,7 +47,7 @@ async def test_async_filtering_sequential_blocks_with_bounded_range(
     initial_block_number = await async_w3.eth.block_number
     builder.toBlock = initial_block_number + 100
     filter_ = await builder.deploy(async_w3)
-    async for i in async_range(100):
```

Comment:
I created the `async_range` helper in a previous commit on this branch because I misunderstood how `for` vs `async for` worked. I thought `async for` was needed if there  were any `await`s in the loop, but it is not.

Reply:
Yeah I can't really figure it out either. It *is* pretty old... I'm assuming there were other test providers running our core test suite in the past. I'm OK taking this out since the whole core suite runs with `EthereumTesterProvider()`.